samlang-core/src/services/__tests__/source-processor.test.ts

```diff
@@ -64,3 +86,108 @@ GLOBAL_STRING_1:
     .text
 `);
 });
+
+it('compile hello world to JS integration test', () => {
+  const moduleReference = new ModuleReference(['Test']);
+  const sourceCode = `
+    class Main {
+        function main(): unit = println('Hello '::'World!')
+    }
+    `;
+  const { checkedSources } = checkSources([[moduleReference, sourceCode]]);
+  const hirSources = compileSamlangSourcesToHighIRSources(checkedSources);
+  expect(highIRSourcesToJSString(hirSources)).toBe(
+    `{const _module_Test_class_Main_function_main = () => {let _t0 = _builtin_throw('dummy');;let _t1 = _builtin_println(0);; }}`
+  );
+});
+
+it('HIR statements to JS string test', () => {
+  expect(
+    highIRStatementToString(
+      HIR_IF_ELSE({
+        booleanExpression: HIR_BINARY({
+          operator: '==',
+          e1: HIR_INT(BigInt(5)),
+          e2: HIR_INT(BigInt(5)),
+        }),
+        s1: [
+          {
+            __type__: 'HighIRReturnStatement',
+          },
+        ],
+        s2: [
+          {
+            __type__: 'HighIRReturnStatement',
+          },
+        ],
+      })
+    )
+  ).toBe(`if (5 == 5) {} else {}`);
+  expect(
+    highIRStatementToString(
+      HIR_FUNCTION_CALL({
+        functionArguments: [],
+        functionExpression: HIR_NAME('func'),
+        returnCollector: 'val',
+      })
+    )
+  ).toBe('let val = func();');
```

Comment:
`let` should be `var`. See https://samlang.developersam.com/docs/intermediate-representation#hir-statements

Reply:
but `var` is bad :/

Reply:
It's bad, but it's necessary. It's the way how HIR lowering works result in this decision:

Consider this if-expression `if (b) then { val _ = println("a");  "a" } else { val _ = println("b");  "b" }`

It is translated into this HIR (written is JS syntax):

```js
if (b) {
 println("a");
 var _temp0 = "a";
} else {
 println("b");
 var _temp0 = "b";
}
_temp0 // as result of this expression
```

You can see that we have to abuse `var`'s function scope capability.